ios/Example/Tests/TrifleTests.swift

```diff
@@ -75,34 +96,125 @@ MIIBZTCCAQqgAwIBAgIBATAKBggqhkjOPQQDAjAYMRYwFAYDVQQDDA1pc3N1aW5nRW50aXR5MB4XDTIz
         
         let mobileCertReq = try trifle.generateMobileCertificateRequest(keyHandle: keyHandle)
-
-        let deviceCertEncoded = Data(base64Encoded: """
-MIIBFDCBvKADAgECAgYBhtoH23kwCgYIKoZIzj0EAwIwEzERMA8GA1\
-UEAwwIY2FzaC5hcHAwHhcNMjMwMzEzMDgxMzEzWhcNMjMwOTA5MDgx\
-MzEzWjARMQ8wDQYDVQQDDAZlbnRpdHkwWTATBgcqhkjOPQIBBggqhk\
-jOPQMBBwNCAASGmNOZtRsOHr2XJVjL1jxRpeumY/jAF3rBDXhhbP49\
-5VYvMYkHfWjeNBzJ1YIqXLQf4BHdQQzDjSvT+abzpdjEMAoGCCqGSM\
-49BAMCA0cAMEQCIBXBwAm0Qn7S4vKRIHYLXDxhUQTqMuReoWBPfiOX\
-fREnAiByNBjSwVxp980/Nl17UvkBUNhTvp55CNuwMmlnporNNA==
-""")!
-
-        let otherRootCertEncoded = Data(base64Encoded: """
-MIHcMIGPoAMCAQICAQEwBQYDK2VwMBgxFjAUBgNVBAMMDWlzc3VpbmdFbnR\
-pdHkwHhcNMjMwMzEzMDYxMzI3WhcNMjMwMzE0MDYxMzI3WjAYMRYwFAYDVQ\
-QDDA1pc3N1aW5nRW50aXR5MCowBQYDK2VwAyEAm+Ac7932SHDPQLYd3p3gm\
-grcArUWrBqhPEC+q/QI3lEwBQYDK2VwA0EAJrfzN7qA3VqwazsT8yMIYMvY\
-Rz2iDA1898Yx5ELtlQcl7QUGXUmadwzW7rpxQB5wIk46tPTEJCFmUIYwrCB\
-4BQ==
-""")
         
         let deviceCertificate = Certificate(version: 0, certificate: deviceCertEncoded)
         let otherRootCertificate = Certificate(version: 0, certificate: otherRootCertEncoded)
+        
         let isVerified = deviceCertificate.verify(
             certificateRequest: mobileCertReq,
-            certificateChain: [],
+            intermediateChain: [],
             rootCertificate: otherRootCertificate
         )
         
         XCTAssertFalse(isVerified)
     }
+    
+    func testSign_succeeds() throws {
+        // NOTE: Right now this test passes because we are not validating
+        // that the certificate contains the public key matching
+        // the signing key
+        // This is TODO - once this is validated, this test should FAIL
+        
+        let data = "hello world".data(using: .utf8)!
+
+        let trifle = try Trifle(reverseDomain: reverseDomain)
+        let keyHandle = try trifle.generateKeyHandle()
+
+        let deviceCertificate = Certificate(version: 0, certificate: deviceCertEncoded)
+        let rootCertificate = Certificate(version: 0, certificate: rootCertEncoded)
+
+        // cert chain of length 2
+        let signData1 = try trifle.createSignedData(data: data,
+                                    keyHandle: keyHandle,
+                                    certificate: [deviceCertificate] + [rootCertificate])
+        XCTAssertNotNil(signData1)
+        
+        // cert chain of length 1
+        let signData2 = try trifle.createSignedData(data: data,
+                                     keyHandle: keyHandle,
+                                     certificate: [deviceCertificate] )
+        XCTAssertNotNil(signData2)
+    }
+    
+    func testSignMultipleKeys_succeeds() throws {
+        // NOTE: Right now this test passes because we are not validating
+        // that the certificate contains the public key matching
+        // the signing key
+        // This is TODO - once this is validated, this test should FAIL
```

Comment:
we can actually move this comment to the assertion statements as a second parameter